src/Tasks/Backup/Zip.php

```diff
@@ -97,13 +97,24 @@ public function add(string | iterable $files, string $nameInZip = null): self
             $files = [$files];
         }
 
+        $compressionMethod = config('backup.backup.destination.compression_method', null);
+        $compressionLevel = config('backup.backup.destination.compression_level', 9);
+
         foreach ($files as $file) {
             if (is_dir($file)) {
                 $this->zipFile->addEmptyDir(ltrim($nameInZip ?: $file, DIRECTORY_SEPARATOR));
             }
 
             if (is_file($file)) {
-                $this->zipFile->addFile($file, ltrim($nameInZip, DIRECTORY_SEPARATOR)).PHP_EOL;
+                $this->zipFile->addFile($file, ltrim($nameInZip ?: '', DIRECTORY_SEPARATOR));
+
+                if ($nameInZip && is_int($compressionMethod)) {
+                    $this->zipFile->setCompressionName(
+                        ltrim($nameInZip, DIRECTORY_SEPARATOR),
```

Comment:
now that I think about it better, maybe
`ltrim($nameInZip ?: '', DIRECTORY_SEPARATOR) ?: $file,`

Reply:
I think so, as we'd rather use compression as well for the case where `$filepath` is used. I'd need to make some more investigation here.